include/sys/util.h

```diff
@@ -262,106 +341,109 @@ uint8_t u8_to_dec(char *buf, uint8_t buflen, uint8_t value);
-
-/* Here's the core trick, we map "_XXXX1" to "_YYYY," (i.e. a string
- * with a trailing comma), so it has the effect of making this a
- * two-argument tuple to the preprocessor only in the case where the
- * value is defined to "1"
- *   ENABLED:    _YYYY,    <--- note comma!
- *   DISABLED:   _XXXX
- */
-#define _XXXX1 _YYYY,
-
-/* Then we append an extra argument to fool the gcc preprocessor into
- * accepting it as a varargs macro.
- *                         arg1   arg2  arg3
- *   ENABLED:   Z_IS_ENABLED3(_YYYY,    1,    0)
- *   DISABLED   Z_IS_ENABLED3(_XXXX 1,  0)
- */
-#define Z_IS_ENABLED2(one_or_two_args) Z_IS_ENABLED3(one_or_two_args true, false)
-
-/* And our second argument is thus now cooked to be 1 in the case
- * where the value is defined to 1, and 0 if not:
- */
-#define Z_IS_ENABLED3(ignore_this, val, ...) val
-
 /**
  * @brief Insert code depending on result of flag evaluation.
  *
- * This is based on same idea as @ref IS_ENABLED macro but as the result of
- * flag evaluation provided code is injected. Because preprocessor interprets
- * each comma as an argument boundary, code must be provided in the brackets.
- * Brackets are stripped away during macro processing.
+ * This is based on same idea as IS_ENABLED(), but as the result of
+ * _flag evaluation either _if_1_code or _else_code is expanded.
  *
- * Usage example:
+ * Because the preprocessor interprets each comma as an argument
+ * boundary, the _if_1_code and _else_code expressions must be inside
+ * brackets/parentheses: (). These are stripped away during macro
+ * expansion.
  *
- * \#define MACRO(x) COND_CODE_1(CONFIG_FLAG, (uint32_t x;), ())
+ * Example:
  *
- * It can be considered as alternative to:
+ *     #define MACRO(x) COND_CODE_1(CONFIG_FLAG, (uint32_t x;), ())
  *
- * \#if defined(CONFIG_FLAG) && (CONFIG_FLAG == 1)
- * \#define MACRO(x) uint32_t x;
- * \#else
- * \#define MACRO(x)
- * \#endif
+ * If CONFIG_FLAG is defined to 1, this expands to:
  *
- * However, the advantage of that approach is that code is resolved in place
- * where it is used while \#if method resolves given macro when header is
- * included and product is fixed in the given scope.
+ *     uint32_t x;
  *
- * @note Flag can also be a result of preprocessor output e.g.
- *	 product of NUM_VA_ARGS_LESS_1(...).
+ * It expands to nothing otherwise.
  *
- * @param _flag		Evaluated flag
- * @param _if_1_code	Code used if flag exists and equal 1. Argument must be
- *			in brackets.
- * @param _else_code	Code used if flag doesn't exists or isn't equal 1.
+ * This could be used as an alternative to:
  *
+ *     #if defined(CONFIG_FLAG) && (CONFIG_FLAG == 1)
+ *     #define MACRO(x) uint32_t x;
+ *     #else
+ *     #define MACRO(x)
+ *     #endif
+ *     MACRO(x)
+ *
+ * However, the advantage of COND_CODE_1() is that code is resolved in
+ * place where it is used, while the \#if method defines MACRO on line,
+ * and requiring it to be invoked again on a separate line. This makes
```

Comment:
~defines MACRO on line and requiring it to be invoked again~

defines MACRO on one line and requires it to be defined again

?